invenio_records_resources/services/records/components/base.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020-2022 CERN.
+# Copyright (C) 2021 Northwestern University.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Records service component base classes."""
+
+from ...base.components import BaseServiceComponent
+
+
+class ServiceComponent(BaseServiceComponent):
+    """Base service component."""
+
+    def create(self, identity, **kwargs):
+        """Create handler."""
+        pass
+
+    def read(self, identity, **kwargs):
+        """Read handler."""
+        pass
+
+    def update(self, identity, **kwargs):
+        """Update handler."""
+        pass
+
+    def delete(self, identity, **kwargs):
+        """Delete handler."""
+        pass
+
+    def search(self, identity, search, params, **kwargs):
+        """Search handler."""
+        return search
+
+
+class BaseRecordFilesComponent(ServiceComponent):
+    _files_attr_key = None
```

Comment:
aligned with @slint 
this is part where we configure record file components on which file field they should refer to: in our case it is between media_files and files